### PR TITLE
[WebNN EP] Remove constraint for conv ops on CPU backend

### PR DESCRIPTION
**Pull request by Honry**
Currently WebNN TFLite backend allows the filter of conv2d/convTranspose2d be an input. Remove the constraint and operate necessary transpose/reshape operations for the filter input.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

PS: We will finally remove preferred layout in future once the Chromium adds transpositions around those ops which are layout sensitive ([conv2d](https://chromium-review.googlesource.com/c/chromium/src/+/5528084?usp=search) has done). Which would make our code much concise.

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 1]
```diff
@@ -168,7 +167,7 @@ Status AddInitializerInNewLayout(ModelBuilder& model_builder,
   if (is_conv == 1)
     dest_shape = {out_t, h_t, w_t, in_t};  // L_0231
   else
-    dest_shape = {in_t, h_t, w_t, out_t};  // L_1230 for depthwise conv weight
+    dest_shape = {in_t, h_t, w_t, out_t};  // L_1230 for depthwise conv and convTranspose weight
 
   SafeInt<size_t> num_elements = SafeInt<size_t>(Product(dest_shape));
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:173]: Existing, but extra break.

```
  size_t element_size{0};
  switch (data_type) {
    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:
      element_size = sizeof(uint8_t);
      break;
    case ONNX_NAMESPACE::TensorProto_DataType_INT8:
      element_size = sizeof(int8_t);
      break;
    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
      element_size = sizeof(uint16_t);
      break;
    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
      element_size = sizeof(float);
      break;
      break; <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
    default:
      break;
  }
```

**Review by fdwr — APPROVED** [review]

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:173]: Good catch!

**Review by fdwr — APPROVED** [review]

**Pull request merged**